examples/Bickley_jet_shallow_water.jl

```diff
@@ -0,0 +1,258 @@
+#
+# FJP: 2] Need to reduce these lines here as this is way too much!
+
+using Oceananigans.Architectures: CPU, architecture
+using Oceananigans.Models: ShallowWaterModel
+using Oceananigans.Grids
+using Oceananigans.Grids: Periodic, Bounded, RegularRectilinearGrid
+using Oceananigans.Grids: xnodes, ynodes, interior
+using Oceananigans.Simulations: Simulation, set!, run!
+using Oceananigans.Coriolis: FPlane
+using Oceananigans.Advection: WENO5
+using Oceananigans.OutputWriters: NetCDFOutputWriter, TimeInterval, IterationInterval
+using Oceananigans.Fields, Oceananigans.AbstractOperations
+
+import Oceananigans.Utils: cell_advection_timescale, prettytime
+
+# ## Two-dimensional domain 
+#
+# The shallow water model is a two-dimensional model and we must specify
+# the number of vertical grid points to be one.  
+# We pick the length of the domain to fit one unstable mode along the channel
+# exactly.  Note that ``Lz`` is the mean depth of the fluid.  It is determined
+# using linear stability theory that the wavenumber that yields the most unstable
+# mode is ``k=0.95``. See [Linear-Stability-Calculators](https://github.com/francispoulin/Linear-Stability-Calculators/tree/main/ShallowWater/Julia/Cartesian)
+# for details on how that is computed.
+
+Lx = 2π/0.95
+Ly = 20
+Lz = 1
+Nx = 128
+Ny = Nx
+
+grid = RegularRectilinearGrid(size = (Nx, Ny, 1),
+                            x = (0, Lx), y = (0, Ly), z = (0, Lz),
+                            topology = (Periodic, Bounded, Bounded))
+
+# ## Physical parameters
+#
+# This is a toy problem and we choose the parameters so this jet idealizes
+# a relatively narrow mesosale jet.   
+# The physical parameters are
+#
+# FJP: 3] Should we change the parameters to be on planetary scales?
+# 
+#   * ``f``: Coriolis parameter
+#   * ``g``: Acceleration due to gravity
+#   * ``U``: Maximum jet speed
+#   * ``\Delta\eta``: Maximum free-surface deformation that is dictated by geostrophy
+#   * ``\epsilon`` : Amplitude of the perturbation
+
+f = 1;
+g = 9.80665;
+U = 1.0;
+Δη = f * U / g;
+ϵ = 1e-4;
+
+# ## Building a `ShallowWaterModel`
+#
+# We use `grid`, `coriolis` and `gravitational_acceleration` to build the model.
+# Furthermore, we specify this runs on `CPUs`, uses `RungeKutta3` for time-stepping
+# and `WENO5` for advection.
+
+model = ShallowWaterModel(
+    architecture=CPU(),
+    timestepper=:RungeKutta3,
+    advection=WENO5(),
+    grid=grid,
+    gravitational_acceleration=g,
+    coriolis=FPlane(f=f),
+    )
+
+# ## Background state and perturbation
+# 
+# We specify `Ω` to be the background vorticity of the jet in the absence of any perturbations.
+# The initial conditions have a small perturbation that is random in space and 
+# decays away from the center of the jet.
+
+  Ω(x, y, z) = 2 * U * sech(y - Ly/2)^2 * tanh(y - Ly/2);
+ uⁱ(x, y, z) =   U * sech(y - Ly/2)^2 .+ ϵ * exp(- (y - Ly/2)^2 ) * randn();
+ ηⁱ(x, y, z) = -Δη * tanh(y - Ly/2);
+ hⁱ(x, y, z) = model.grid.Lz .+ ηⁱ(x, y, z);
```

Comment:
There are functions so no `.+` is needed; simply `+`.

Reply:
Thanks